google/cloud/internal/async_streaming_write_rpc_timeout_test.cc

```diff
@@ -0,0 +1,331 @@
+
+  auto constexpr kStartTimeout = std::chrono::milliseconds(2345);
+  auto constexpr kWriteTimeout = std::chrono::milliseconds(2345);
+
+  auto mock_cq = std::make_shared<MockCompletionQueueImpl>();
+  EXPECT_CALL(*mock_cq,
+              MakeRelativeTimer(std::chrono::nanoseconds(kStartTimeout)))
+      .WillOnce([&](auto) {
+        return sequencer.PushBack("MakeRelativeTimer")
+            .then(
+                [](auto f) -> StatusOr<std::chrono::system_clock::time_point> {
+                  if (!f.get()) return CancelledTimer();
+                  return make_status_or(std::chrono::system_clock::now());
+                });
+      });
+
+  CompletionQueue cq(mock_cq);
+
+  TestedStream uut(cq, kStartTimeout, kWriteTimeout, std::move(mock));
+  auto result = uut.Start();
+  auto timer = sequencer.PopFrontWithName();
+  auto start = sequencer.PopFrontWithName();
+  ASSERT_THAT(timer.second, "MakeRelativeTimer");
+  ASSERT_THAT(start.second, "Start");
+  start.first.set_value(true);
+  timer.first.set_value(false);
+  EXPECT_TRUE(result.get());
+}
+
+TEST(AsyncStreamingWriteRpcTimeout, StartTimeout) {
+  AsyncSequencer<bool> sequencer;
+  auto mock = std::make_unique<MockStream>();
+  EXPECT_CALL(*mock, Start).WillOnce([&] {
+    return sequencer.PushBack("Start");
+  });
+  EXPECT_CALL(*mock, Cancel).Times(1);
+
+  auto constexpr kStartTimeout = std::chrono::milliseconds(2345);
+  auto constexpr kWriteTimeout = std::chrono::milliseconds(2345);
+
+  auto mock_cq = std::make_shared<MockCompletionQueueImpl>();
+  EXPECT_CALL(*mock_cq,
+              MakeRelativeTimer(std::chrono::nanoseconds(kStartTimeout)))
+      .WillOnce([&](auto) {
+        return sequencer.PushBack("MakeRelativeTimer")
+            .then(
+                [](auto f) -> StatusOr<std::chrono::system_clock::time_point> {
+                  if (!f.get()) return CancelledTimer();
+                  return make_status_or(std::chrono::system_clock::now());
+                });
+      });
+
+  CompletionQueue cq(mock_cq);
+
+  TestedStream uut(cq, kStartTimeout, kWriteTimeout, std::move(mock));
+  auto result = uut.Start();
+  auto timer = sequencer.PopFrontWithName();
+  auto start = sequencer.PopFrontWithName();
+  ASSERT_THAT(timer.second, "MakeRelativeTimer");
+  ASSERT_THAT(start.second, "Start");
+  timer.first.set_value(true);
+  start.first.set_value(true);
+  EXPECT_FALSE(result.get());
+}
+
+TEST(AsyncStreamingWriteRpcTimeout, WriteSuccess) {
+  AsyncSequencer<bool> sequencer;
+  auto mock = std::make_unique<MockStream>();
+  EXPECT_CALL(*mock, Write(42, _)).WillOnce([&] {
+    return sequencer.PushBack("Write");
+  });
+  EXPECT_CALL(*mock, Cancel).Times(0);
+
+  auto constexpr kStartTimeout = std::chrono::milliseconds(2345);
+  auto constexpr kWriteTimeout = std::chrono::milliseconds(2345);
+
+  auto mock_cq = std::make_shared<MockCompletionQueueImpl>();
+  EXPECT_CALL(*mock_cq,
+              MakeRelativeTimer(std::chrono::nanoseconds(kStartTimeout)))
```

Comment:
nit: they are the same ignored value, but I think you want s/kStartTimeout/kWriteTimeout/

Reply:
The values should have been different. Fixed.